### PR TITLE
ceph: use absolute binary paths instead of relative paths

### PR DESCRIPTION
**Pull request by NukaDuka**
## Description of changes

- **ceph**: While trying to mount CephFS using libceph and systemd, `mount.ceph` tries to call `modinfo`, `modprobe`, and `grep`, but fails with the error `sh: line 1: modprobe: command not found`. This is because ceph calls these binaries by running the command `sh -c -- '<application> %s %s'`, which does not pass the `PATH` environment variable through. This isn't usually a problem, because ceph, by default, calls the paths of these binaries as they would be in debian, in `/sbin` and `/bin`, but a change was made to replace these with relative paths, thus breaking the mounting process entirely. Replacing these relative paths with absolute store paths alleviates this issue whilst preserving all functionality.

- **nixosTests.ceph-single-node**: Due to an [issue](https://www.spinics.net/lists/ceph-users/msg77812.html) with the cryptography python library, Ceph Dashboard and other mgr modules are currently broken, which will cause this test to always fail. Removing the check resolves this issue, and brings the test in line with the other Ceph tests, which do not contain the dashboard check.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NukaDuka** [comment]
Result of `nixpkgs-review pr 270444` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>ceph</li>
    <li>ceph-client</li>
    <li>ceph-csi</li>
    <li>ceph.dev</li>
    <li>ceph.doc</li>
    <li>libceph (ceph.lib ,libceph.dev ,libceph.doc ,libceph.lib ,libceph.man)</li>
    <li>ceph.man</li>
    <li>qemu_full</li>
    <li>qemu_full.debug</li>
    <li>qemu_full.ga</li>
    <li>samba4Full</li>
    <li>samba4Full.dev</li>
    <li>samba4Full.man</li>
  </ul>
</details>

**Review by illustris — APPROVED** [review]
```
$ nix-shell -p nixpkgs-review --run "nixpkgs-review pr 270444"
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/270444/head:refs/nixpkgs-review/1
$ git worktree add /home/illustris/.cache/nixpkgs-review/pr-270444-1/nixpkgs d996452779de10baca8631d95e6d689be28c317d
Preparing worktree (detached HEAD d996452779de)
Updating files: 100% (38414/38414), done.
HEAD is now at d996452779de Merge pull request #272825 from r-ryantm/auto-update/dwarfs
$ git merge --no-commit --no-ff 5b0b54d7674e57608294a4bfc1cf8e6273864f94
Automatic merge went well; stopped before committing as requested
$ nix build --nix-path nixpkgs=/home/illustris/.cache/nixpkgs-review/pr-270444-1/nixpkgs nixpkgs-overlays=/run/user/1000/tmp_3u42fdr --extra-experimental-features nix-command no-url-literals
--no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed -f /home/illustris/.cache/nixpkgs-review/pr-270444-1/build.nix

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/270444

13 packages built:
ceph ceph-client ceph-csi ceph.dev ceph.doc libceph ceph.man qemu_full qemu_full.debug qemu_full.ga sambaFull sambaFull.dev sambaFull.man
```

File: nixos/tests/ceph-single-node.nix [hunk 1]
```diff
@@ -181,17 +181,6 @@ let
     monA.wait_until_succeeds("ceph osd stat | grep -e '3 osds: 3 up[^,]*, 3 in'")
     monA.wait_until_succeeds("ceph -s | grep 'mgr: ${cfg.monA.name}(active,'")
     monA.wait_until_succeeds("ceph -s | grep 'HEALTH_OK'")
-
-    # Enable the dashboard and recheck health
```
[the remark this answers is not in the document]
Comment (NukaDuka) [reply nixos/tests/ceph-single-node.nix:185]: That makes sense. Fixed

**Review by illustris — APPROVED** [review]

File: pkgs/tools/filesystems/ceph/default.nix [hunk 2]
```diff
@@ -294,10 +296,13 @@ in rec {
...
+    # replace /sbin and /bin based paths with direct nix store paths
+    # increase the `command` buffer size since 2 nix store paths cannot fit within 128 characters
     preConfigure =''
-      substituteInPlace src/common/module.c --replace "/sbin/modinfo"  "modinfo"
-      substituteInPlace src/common/module.c --replace "/sbin/modprobe" "modprobe"
-      substituteInPlace src/common/module.c --replace "/bin/grep" "grep"
+      substituteInPlace src/common/module.c --replace "char command[128];" "char command[256];" \
+                                            --replace "/sbin/modinfo"  "${kmod}/bin/modinfo" \
+                                            --replace "/sbin/modprobe" "${kmod}/bin/modprobe" \
+                                            --replace "/bin/grep" "${gnugrep}/bin/grep"
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/ceph/default.nix:305]:
```suggestion
      substituteInPlace src/common/module.c \
        --replace "char command[128];" "char command[256];" \
        --replace "/sbin/modinfo"  "${kmod}/bin/modinfo" \
        --replace "/sbin/modprobe" "${kmod}/bin/modprobe" \
        --replace "/bin/grep" "${gnugrep}/bin/grep"
```

**Review by lheckemann — APPROVED** [review]

**Pull request merged**